enhancements/network/configure-ovs-alternative.md

```diff
@@ -0,0 +1,498 @@
+runs. However, the platform-specific part can be much simpler because it
+won't need to handle every possible network configuration, just enough to
+reach the Ignition endpoint. In some use cases, platform-specific configuration
+won't even be needed.
+
+### Workflow Description
+
+When the user is populating install-config they will provide the necessary
+configuration in the networking section (TODO: Is this the right place?).
+It will look something like this:
+
+networking:
+  networkType: OVNKubernetes
+  machineNetwork:
+  - cidr: 192.0.2.0/24
+  [...]
+  hostConfig:
+  - name: master-0
+    networkConfig:
+      interfaces:
+      - name: enp2s0
+        type: ethernet
+        state: up
+        ipv4:
+          enabled: false
+        ipv6:
+          enabled: false
+      - name: br-ex
+        type: ovs-bridge
+        state: up
+        copy-mac-from: enp2s0
+        ipv4:
+          enabled: false
+          dhcp: false
+        ipv6:
+          enabled: false
+          dhcp: false
+        bridge:
+          port:
+          - name: enp2s0
+          - name: br-ex
+      - name: br-ex
+        type: ovs-interface
+        state: up
+        ipv4:
+          enabled: true
+          dhcp: true
+        ipv6:
+          enabled: false
+          dhcp: false
+
+The name field will need to be something that uniquely identifies a given host.
+I'm unsure if name is necessarily the best way to do that, but it is one valid
+option. I reused the networkConfig name from the baremetal feature for
+consistency. Perhaps something different would be preferable to avoid confusion
+though?
+
+Note that although the configuration can be provided on a per-host basis, it is
+not mandatory to do so. YAML anchors can be used to duplicate a single
+configuration across multiple nodes. For example:
+
+networking:
+  networkType: OVNKubernetes
+  machineNetwork:
+  - cidr: 192.0.2.0/24
+  [...]
+  hostConfig:
+  - name: master-0
+    networkConfig: &BOND
+      interfaces:
+      - name: br-ex
+        ...etc...
+  - name: master-1
+    networkConfig: *BOND
+  - name: master-2
+    networkConfig: *BOND
+
+This reduces duplication in install-config and the chance of errors due to
+unintended differences in node config.
+
+When this mechanism is in use, a flag will be set to indicate to configure-ovs
+that it should not attempt to configure br-ex.
+
+If changes to the bridge configuration are desired on day 2 (e.g. adding a VLAN
+or changing MTU), the user will need to install Kubernetes-NMState and use it
+to apply those changes directly to br-ex.
+
+#### Variation [optional]
+
+This could be implemented almost entirely with machine-configs, with only a simple
+service on the host to apply the provided NMState files. This is the approach
+the [prototype implementation](https://docs.google.com/document/d/1Zp1J2HbVvu-v4mHc9M594JLqt6UwdKoix8xkeSzVL98/edit#heading=h.fba4j9nvp0nl)
+used since per-host configuration was not available any other way. However,
```

Comment:
- Machine-configs are applied by ignition, the same as this.
- That's a good point. If you scaled out a machine config pool on day 2 you'd have to reboot all of the nodes in that pool because you'd have to add the new node configs to all of them. That's a pretty big limitation.
- True, although in this case I envision the machine-configs writing NMState YAML instead of raw nmconnection files, so you'd still have NMState validating them before it applies them and rolling back if they break things. You couldn't catch syntax errors in install-config though because the installer would just see opaque machine-configs, not their contents.

Actually, the scaleout case is something we need to consider. At install time we can handle mapping things in the installer, but on day 2 we'll need a manual process, much like baremetal has. Unfortunately I don't think we can use kubernetes-nmstate for scaleout either because it won't be running soon enough to apply the bridge config. :-/

Reply:
Note that scaleout works fine with my latest machine-config prototype. You just update the MC with the new node configs and it gets rolled out reboot-less so it will be there when the new nodes start. I discuss how this works below now.